pipeline-steps/templates/steps/vault-upload.yaml

```diff
@@ -14,12 +14,12 @@ steps:
         seachedSecret=$(az keyvault secret list --vault-name ${{ parameters.azureVault }}  --query "[?name=='${{ parameters.secretName }}'].name" -o tsv)
         if [ -z "$seachedSecret" ]; then
               echo  "This is a new secret"
-              az keyvault secret set --name ${{ parameters.secretName }} --vault-name ${{ parameters.azureVault }}  --value '${{ parameters.secretValue }}'
```

Comment:
double quotes better than single quotes for handling special characters, output set to none to avoid exposing the secret.